src/v/cloud_storage/topic_manifest.cc

```diff
@@ -388,15 +388,15 @@ void topic_manifest::serialize(std::ostream& out) const {
     if (!_topic_config->properties.retention_bytes.is_disabled()) {
         w.Key("retention_bytes");
         if (_topic_config->properties.retention_bytes.has_optional_value()) {
-            w.Int64(_topic_config->properties.retention_bytes.value());
+            w.Uint64(_topic_config->properties.retention_bytes.value());
         } else {
             w.Null();
         }
     }
     if (!_topic_config->properties.retention_duration.is_disabled()) {
         w.Key("retention_duration");
         if (_topic_config->properties.retention_duration.has_optional_value()) {
-            w.Int64(
```

Comment:
I'm not sure we should cast a chrono type that is signed to an unsigned.

kinda related:
https://github.com/redpanda-data/redpanda/issues/13432
https://github.com/redpanda-data/redpanda/blob/d6f608ae81d2bd0a148fa4455323505006144d29/src/v/config/property.h#L896-L901

retention_duration_property::to_json serializes infinite retention as -1, and rjson_serilize casts it as unsigned, turning a negative value in a 2^63.

I don't think these two pieces of code interact, so the data serialized by one is not read by the other (otherwise we should have seen bug reports), but worth doing a check

Reply:
Good point. Hopefully we don't ever build negative durations. I wonder if we should clamp to zero?

Reply:
Ah great catch! I updated to pass through negatives.

I'm not against clamping, but I do think the serialization code shouldn't make such heavy-handed decisions: those kinds of decisions probably best left to the caller.